components/fxa-client/src/internal/error.rs

```diff
@@ -148,6 +148,19 @@ error_support::define_error_conversions! {
     }
 }
 
+// Adapts errors from the sync15_traits crate into sync15 errors.
```

Comment:
It's kind of sad that consumers of `sync15` need to handle both `sync15::Error` and `sync15::SyncTraitsError`, but I can't think of a better way to handle it without changing a lot of code.  I think the docstring could be improved though, what about adding someting like "This is needed because `sync15` re-exports functions from `sync15_traits` and those functions return `Result<T, SyncTraitsError>`"